File: increasingring/src/main/java/com/baybaka/increasingring/config/SoundStateDTO.java
package com.baybaka.increasingring.config;

public class SoundStateDTO {

    private int originalRingVolume = -1;
    private int originalRingMode = -1 ;
    private int originalMusicVolume = -1;
    private int originalNotificationVolume = -1;

    public int getOriginalNotificationVolume() {
        return originalNotificationVolume;
    }

    public void setOriginalNotificationVolume(int originalNotificationVolume) {
        this.originalNotificationVolume = originalNotificationVolume;
    }


    public int getOriginalRingVolume() {
        return originalRingVolume;
    }

    public void setOriginalRingVolume(int originalRingVolume) {
        this.originalRingVolume = originalRingVolume;
    }

    public int getOriginalRingMode() {
        return originalRingMode;
    }

    public void setOriginalRingMode(int originalRingMode) {
        this.originalRingMode = originalRingMode;
    }

    public int getOriginalMusicVolume() {
        return originalMusicVolume;
    }

    public void setOriginalMusicVolume(int originalMusicVolume) {
        this.originalMusicVolume = originalMusicVolume;
    }

    @Override
    public String toString() {
        return "SoundStateDTO{" +
                "originalRingVolume=" + originalRingVolume +
                ", originalRingMode=" + originalRingMode +
                ", originalMusicVolume=" + originalMusicVolume +
                ", originalNotificationVolume=" + originalNotificationVolume +
                '}';
    }

    //ignores notification??
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        SoundStateDTO that = (SoundStateDTO) o;

        return originalRingVolume == that.originalRingVolume &&
                originalRingMode == that.originalRingMode &&
                originalMusicVolume == that.originalMusicVolume;

    }

    @Override
    public int hashCode() {
        int result = originalRingVolume;
        result = 31 * result + originalRingMode;
        result = 31 * result + originalMusicVolume;
        return result;
    }
}

File: increasingring/src/main/java/com/baybaka/increasingring/config/SoundStateDTO.kt
package com.baybaka.increasingring.config

import com.baybaka.increasingring.audio.RingMode

class SoundStateDTO {

    var ringVolume = -1
    lateinit var ringMode: RingMode

    var musicVolume = -1

    var notificationVolume = -1

    override fun toString(): String {
        return "SoundStateDTO{" +
                "ringMode=" + ringMode +
                ", ringVolume=" + ringVolume +
                ", musicVolume=" + musicVolume +
                ", notificationVolume=" + notificationVolume +
                '}'
    }

    //ignores notification??
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
//        if (other == null || javaClass != other.javaClass) return false

        val that = other as? SoundStateDTO

        return that?.let {
            ringMode === it.ringMode &&
                    ringVolume == it.ringVolume &&
                    musicVolume == it.musicVolume
        } ?: false


    }

    override fun hashCode(): Int {
        var result = ringVolume
        result = 31 * result + ringMode.ordinal
        result = 31 * result + musicVolume
        return result
    }
}
